### PR TITLE
Better error provenance for stuff invoking withResponse'

### LINKED ISSUE #1185: Yesod Tests addToken could use a nicer error
Getting 

```
There was no response, you should make a request
```

Because you forgot to `GET` before using `addToken` in a POST is misery itself.

### PR DESCRIPTION
**Pull request by bitemyapp**
I tried to take a light touch, mostly mucking with the prime variants of functions.

Here's how it looks:

screenshot from 2016-03-17 14-35-08

This is intended to resolve #1185

### CODE REVIEW

**Comment by codygman** [comment]
:+1: Seems like an improvement to me.

**Comment by bitemyapp** [comment]
The test failure seems spurious, could be wrong.

File: yesod-test/yesod-test.cabal [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name:               yesod-test
-version:            1.5.0.1
+version:            1.6.0.0
```
Comment (snoyberg) [inline yesod-test/yesod-test.cabal]: What's the reason for the major version bump? It looks like this only changes internal functions.

Comment (bitemyapp) [inline yesod-test/yesod-test.cabal]: Oh so it does. I'd managed that on accident, thinking I'd changed the type signature of something that _was_ exported. Nuts. I'll fix it, sorry!

**Comment by snoyberg** [comment]
Thanks Chris!

**Pull request merged**